narwhals/_pandas_like/series.py

```diff
@@ -704,6 +704,11 @@ class PandasLikeSeriesDateTimeNamespace:
     def __init__(self, series: PandasLikeSeries) -> None:
         self._pandas_series = series
 
+    def date(self) -> PandasLikeSeries:
+        return self._pandas_series._from_native_series(
+            self._pandas_series._native_series.dt.date,
+        )
```

Comment:
I wonder what are the pros and cons of returning a series of type `datetime64[ns]`.

Polars returns a `Date` type which we can continue to chain. Example from the docstring you added:

```py
s_pl.dt.date().dt.year()
```

Reply:
Good point! 
Not sure how to go about from this point on... I think this would be the same with  the existing other `dt` methods (?)
Maybe it will require to have a custom class to handle dt chainign on pandas?

Reply:
thanks @FBruzzesi for observing this!

I think we should only support this if the native series is backed by a pyarrow dtype (e.g. `'timestamp[ns][pyarrow]'`), so that pandas gives us a `date32[day][pyarrow]` output, and otherwise raise notimplementederror

to make the check easier across libraries, maybe assign `.dt.date` to some variable `result`, then check if `result.dtype == object`, and if so, raise not implemented, advising to use pyarrow-backed dtypes?